plugin/cache/cache_test.go

```diff
@@ -17,124 +17,129 @@ import (
 )
 
 type cacheTestCase struct {
-	test.Case                    // the expected message coming "out" of cache
-	in                 test.Case // the test message going "in" to cache
-	AuthenticatedData  bool
-	RecursionAvailable bool
-	Truncated          bool
-	shouldCache        bool
+	out         test.Case // the expected message coming "out" of cache
+	in          test.Case // the test message going "in" to cache
+	shouldCache bool
```

Comment:
At first glance, it does seem cleaner to move `AuthenticatedData`, `RecursionAvailable`, and `Truncated` into `test.Case` and to have an explicit expected `out` value for each test case.  However, it seems that `TestCacheInsertion` doesn't actually check these fields in `out`:  I can remove everything except `Answer` or `Ns` from `out` in each of the test cases, and tests still pass.  That seems strange to me.

Reply:
I also realize this PR leaves things in a bit of a transitional phase in which these fields are move into `test.Case`, but other unit tests are still utilizing adding the header bits at the top level test case structure such as in `cname_test.go`:
https://github.com/coredns/coredns/blob/2fe5890d8b7fc7801d308dbb51bf29c57ee9a92b/plugin/template/cname_test.go#L66

I don't think this is too problematic to be in this transitional state, since I feel like it like the header flags should be transitioned to be specified in the `test.Case`, but feel free to let me know if you feel differently.